packages/ember-data/lib/serializers/json-api-serializer.js

```diff
@@ -8,7 +8,91 @@ import { pluralize, singularize } from 'ember-inflector/lib/system/string';
 
 var dasherize = Ember.String.dasherize;
 
+/*
+  Ember Data 2.0 Serializer:
+
+  In Ember Data a Serializer is used to serialize and deserialize
+  records when they are transferred in and out of an external source.
+  This process involves normalizing property names, transforming
+  attribute values and serializing relationships.
+
+  `JSONApiSerializer` supports the http://jsonapi.org/ spec and is the
+  serializer recommended by Ember Data.
+
+  This serializer normalizes a JSONApi payload that looks like:
+
+  ```js
+    App.Player = DS.Model.extend({
+      name: DS.attr(),
+      skill: DS.attr(),
+      gamesPlayed: DS.attr(),
+      club: DS.belongsTo('club')
+    });
+    App.Club = DS.Model.extend({
+      name: DS.attr(),
+      location: DS.attr(),
+      players: DS.hasMany('player')
+    });
+  ```
+  ```js
+    {
+      "data": [
+        {
+          "attributes": {
+            "name": "Benfica",
+            "location": "Portugal"
+          },
+          "id": "1",
+          "relationships": {
+            "players": {
+              "data": [
+                {
+                  "id": "3",
+                  "type": "players"
+                }
+              ]
+            }
+          },
+          "type": "clubs"
+        }
+      ],
+      "included": [
+        {
+          "attributes": {
+            "name": "Eusebio Silva Ferreira",
+            "skill": "Rocket shot",
+            "games-played": 431
+          },
+          "id": "3",
+          "relationships": {
+            "club": {
+              "data": {
+                "id": "1",
+                "type": "clubs"
+              }
+            }
+          },
+          "type": "players"
+        }
+      ]
+    }
+  ```
+  to the format that the Ember Data store expects.
+
+  @class JSONApiSerializer
+  @namespace DS
+  @extends DS.JSONSerializer
+*/
+
 /**
+  In Ember Data a Serializer is used to serialize and deserialize
```

Comment:
This comment block and the one above can get merged. Otherwise the last one (below) will win.